### PR TITLE
Allows mechs to pick up trash piles and salvageable

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request
Mechs really should be able to load up a bunch of junk and move it around with a mech

## Changelog
:cl:
/:cl:

### CODE REVIEW

**Comment by Kazkin** [comment]
Gonna let Lessie take a look just in case.

**Comment by KathyRyals** [comment]
DNR until properly tested.

File: code/game/mecha/equipment/tools/tools.dm [hunk 1]
```diff
@@ -49,7 +50,7 @@
 			if(locate(/mob/living) in O)
 				occupant_message(SPAN_WARNING("You can't load living things into the cargo compartment."))
 				return
-			if(O.anchored)
+			if(O.anchored &! allow_pickup)
```
Comment (KathyRyals) [inline code/game/mecha/equipment/tools/tools.dm:53]: This hurts so much I had to call someone else to review it for me.

**Review by KathyRyals — CHANGES_REQUESTED** [review]

File: code/game/mecha/equipment/tools/tools.dm (hunk 1 above)
Comment (Timberpoes) [reply code/game/mecha/equipment/tools/tools.dm:53]: Trilbyspaceclone, what you've just coded is one of the most insanely idiotic things I have ever read. At no point in your rambling, incoherent pull request were you even close to anything that could be considered a functional piece of code. Everyone in this GitHub repo is now dumber for having seen it. I award you no points, and may God have mercy on your soul.

Comment (TiviPlus) [reply code/game/mecha/equipment/tools/tools.dm:53]: I was shown a screenshot of this and as the person that largely rewrote mechs for tg my eyes are now stained why would you do  this to me

Comment (Timberpoes) [inline code/game/mecha/equipment/tools/tools.dm:53]: You are attempting to bitwise AND a list() with a boolean value.

Here's some quick half-assed code to showcase *A* method to accomplish what you want.
```
if(O.anchored && (!istype(O, /obj/structure/scrap) && !istype(O, /obj/structure/salvageable))
```

Alternatively, just get out of your mech and unanchor the scrap, if that's even possible?

Comment (Trilbyspaceclone) [reply code/game/mecha/equipment/tools/tools.dm:53]: Thx, tryed to do that at the start and kinda failed at it

**Pull request merged**